WordPress/Classes/ViewRelated/Me/AccountSettingsViewController.swift

```diff
@@ -83,11 +96,44 @@ private struct AccountSettingsController: SettingsController {
             ])
     }
 
+    
+    // MARK: - Helpers
+    
+    func blogWithBlogId(blogId: Int?) -> Blog? {
+        let service = BlogService(managedObjectContext: context)
+        return service.blogByBlogId(blogId)
+    }
+    
+    func updatePrimarySite(primarySiteObjectId: NSManagedObjectID) {
+        guard let newPrimarySite = try! context.existingObjectWithID(primarySiteObjectId) as? Blog else {
+            return
+        }
+        let change = AccountSettingsChange.PrimarySite(newPrimarySite.dotComID as Int)
+        self.service.saveChange(change)
+    }
+    
+    
     // MARK: - Actions
 
     func editWebAddress() -> ImmuTableRowControllerGenerator {
         return editText(AccountSettingsChange.WebAddress, hint: NSLocalizedString("Shown publicly when you comment on blogs.", comment: "Help text when editing web address"))
     }
+    
+    func editPrimarySite(primarySite: Blog?) -> ImmuTableRowControllerGenerator {
+        return {
+            row in
+
+            let selectorViewController = BlogSelectorViewController(selectedBlogObjectID: primarySite?.objectID,
+                selectedCompletion: { self.updatePrimarySite($0) },
```

Comment:
The magic of `struct` :smile:. Nevertheless, thanks for spotting that. Would have been a nasty retain cycle!